lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2204,8 +2204,55 @@ namespace Js
     {
         charcount_t count = pThis->GetLength();
 
+        const char16* inStr = pThis->GetString();
+        const char16* inStrLim = inStr + count;
+        const char16* i = inStr;
+
+        // Try to find out the chars that do not need casing (in the ASCII range)
+        if (toCase == ToUpper)
+        {
+            while (i < inStrLim)
+            {
+
+                // first range of ascii lower-case (97-122)
+                // second range of ascii lower-case (223-255)
+                // non-ascii chars (255+)
+                if ('a' <= *i)
+                {
+                    if (*i <= 'z') { break; }
+                    if ('�' <= *i) { break; }
```

Comment:
Why the change from
`if (('a' <= *i && *i <= 'z') || *i >= 'ß') { break; }`

I apologize, my comment earlier about the `<=` style was meant only for when you are comparing a variable with a range inline in one expression.  This is meant to be more readable because it looks like math, e.g. `a ≤ i ≤ b`.

`('a' <= *i && *i <= 'z') || *i >= 'ß'`

This reads nicely like "when `*i` is between `'a'` and `'z'`, or `*i` is at or over `'ß'`.

When broken out in multiple if statements like your current code above I would prefer the `>=` form like this for readability:

`if (*i > = 'a') { ... }`

That reads like "when `*i` is at or above `'a'`" which is easier than "when `'a'` is the same or below `*i`".